dvc/repo/metrics/show.py

```diff
@@ -123,20 +116,47 @@ def _collect_metrics(self, path, recursive, typ, xpath, branch):
     return res
 
 
+def _read_metrics_filesystem(self, path, typ, xpath):
+    if not os.path.exists(path):
+        return None
+    with open(path, "r") as fd:
+        return _read_metric(self, fd, typ=typ, xpath=xpath)
+
+
 def _read_metrics(self, metrics, branch):
     res = {}
     for out, typ, xpath in metrics:
-        assert out.scheme == "local"
-        if out.use_cache:
-            path = self.cache.local.get(out.checksum)
-        else:
-            path = out.path
-
-        metric = _read_metric(path, typ=typ, xpath=xpath, branch=branch)
-        if not metric:
-            continue
-
-        res[out.rel_path] = metric
+        try:
+            assert out.scheme == "local"
```

Comment:
Does it needs to be ensured in case if `out.use_cache` is true?

Reply:
yes, I think so. We don't support external metrics (like when you write a metric file to HDFS). There is a ticket for this I believe: https://github.com/iterative/dvc/issues/1664